### PR TITLE
[BE] [ADD] add stripe connectivity 

### PR DESCRIPTION
**Pull request by MarcPIster**
### 🛠 Changes being made

At some point we want to make money. 
To do this, we need a way to collect money.
also added simple page for showing success and failed status.

### ✨ What's the context?

Added stripe connectivity.

### 🧪 Test plan

You need to add the following two lines to the .env in the backend:
STRIPE_SECRET_KEY=""
FREE_PAYMENT=""
(For these values: https://epitech-eip.atlassian.net/wiki/spaces/EIP/pages/10452993/DataBase+Access+.env+file)

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/backend/src/routes/payments.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const router = express.Router();
+
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY as string, {
+  apiVersion: '2023-10-16',
+});
+
+router.post('/create-checkout-session',
+  async (req: express.Request, res: express.Response) => {
+    try {
+      console.log(process.env.STRIPE_SECRET_KEY);
```
Comment (JosefineM2000) [inline packages/backend/src/routes/payments.ts:14]: please remove this log

File: packages/backend/src/routes/payments.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        mode: 'payment',
+        // eslint-disable-next-line camelcase
+        success_url: `${domainURL}/payment-success`,
+        // eslint-disable-next-line camelcase
+        cancel_url: `${domainURL}/payment-failed`,
+      });
+
+      res.redirect(303, session.url);
+    } catch (error) {
+      console.error('Fehler beim Erstellen der Checkout-Session:', error);
```
Comment (JosefineM2000) [inline packages/backend/src/routes/payments.ts:39]: please translate to english

File: packages/visitorWeb/src/pages/Payments/Failed/PaymentFailedContainer.module.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
+.paymentFailedContainer {
+  text-align: center;
+  padding: 50px;
+  max-width: 600px;
+  margin: auto;
+
+  h2 {
+    color: #F44336; // Eine Farbe für Fehler
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/Payments/Failed/PaymentFailedContainer.module.scss:8]: please either remove or translate comment

File: packages/backend/src/routes/payments.ts
Comment (JosefineM2000) [inline packages/backend/src/routes/payments.ts:1]: Please add tests.

**Review by JosefineM2000 — APPROVED** [review]
